apollo-api/src/main/java/com/spotify/apollo/RequestValue.java

```diff
@@ -88,7 +88,9 @@ public Request withHeader(String name, String value) {
   @Override
   public Request withHeaders(Map<String, String> additionalHeaders) {
     Map<String, String> headers = new LinkedHashMap<>(headers());
-    headers.putAll(additionalHeaders);
+    for (Map.Entry<String, String> header : additionalHeaders.entrySet()) {
+      headers.put(header.getKey().toLowerCase(), header.getValue());
+    }
```

Comment:
Is the right approach? This would effectively tie this class to only be usable with protocols where case-insensitivity is required for headers.